src/main/java/org/sopt/lequuServer/domain/book/service/BookService.java

```diff
@@ -27,4 +33,51 @@ public BookCreateResponseDto createBook(Book book, Member member) {
 
         return BookCreateResponseDto.of(createdBook);
     }
+
+    // 정해진 기간의 레큐노트를 가져와 많이 부착된 순서로 레큐북 정렬하는 로직
+    public List<Book> sortBooksDesc(LocalDateTime startDate, LocalDateTime endDate) {
+        return bookRepository.findBooksMonth(startDate, endDate)
+            .stream()
+            .sorted(Comparator.comparingInt(book -> -book.getNotes().size())) // -를 붙여 내림차순 정렬
+            .collect(Collectors.toList());
+    }
+
+    // 최근 3달을 확인하는데 최근 순으로 먼저 인기레큐북에 추가할 수 있는 로직
+    // 현재를 0이라 하면 -1 ~ 0(최근 한달), -2 ~ -1(지지난달 부터 지난달), -3 ~ -2(지지지난달 부터 지지난달)
+    public List<Book> getPopularBooks() {
+        List<Book> popularBooks = new ArrayList<>();
+        LocalDateTime today = LocalDateTime.now();
+
+        // 최근 3달 확인
+        for (int i = 0; i < 3; i++) {
+            LocalDateTime endDate = today.minusMonths(i);
+            LocalDateTime startDate = endDate.minusMonths(1);
+            List<Book> books = sortBooksDesc(startDate, endDate);
+
+            if (popularBooks.size() + books.size() <= 6) {  // 만약 현재있는 인기북과 가져온 북이 6이하면 인기북에 가져온 것 다 추가
+                popularBooks.addAll(duplicateBook(books, popularBooks));
+            } else {
+                int remain = 6 - popularBooks.size();
+                popularBooks.addAll(duplicateBook((books.subList(0, remain)), popularBooks));
+            } // 예를 들어 인기북에 2개가 있으면 4개만 더 추가하면 되니까 가져온 북의 인덱스 0부터 3까지만 가져오도록 함
+
+            if (popularBooks.size() >= 6 || i == 2) {
+                break;
+            } // 인기북이 6개 이상이 되거나 최근 3달을 다 검사했으면 for문을 빠져나옴
+        }
+
+        return popularBooks;
+    }
+
+    // 인기레큐북에 중복된 레큐북을 방지하기 위한 로직
+    public List<Book> duplicateBook(List<Book> books, List<Book> popularBooks) {
+        List<Book> duplicatedBook = new ArrayList<>();
+
+        for (Book book : books) {
+            if (!popularBooks.contains(book)) {
+                duplicatedBook.add(book);
+            }
+        }
+        return duplicatedBook;
```

Comment:
```suggestion
        return popularBooks.stream().filter(o -> books.stream()
				.noneMatch(Predicate.isEqual(o))).collect(Collectors.toList());
```
[P3] 동일한 객체끼리 비교하는 로직이면 stream으로 위와 같이 작성할 수도 있어요!
이렇게 하면 popularBooks에 중복되지 않는 book 리스트만 남아서 return됩니다!
이것도 위에 것처럼 참고로 봐주세요~~

(예시)
List<String> oldList = Arrays.asList("1", "2", "3", "4");
		List<String> newList = Arrays.asList("3", "4", "5", "6");

		List<String> oldNoneMatchList = oldList.stream().filter(o -> newList.stream()
				.noneMatch(Predicate.isEqual(o))).collect(Collectors.toList());

		List<String> newNoneMatchList = newList.stream().filter(n -> oldList.stream()
				.noneMatch(Predicate.isEqual(n))).collect(Collectors.toList());

		System.out.println(oldNoneMatchList.toString());		// [1,2]
		System.out.println(newNoneMatchList.toString());		// [5,6]

Reply:
👍👍👍🔥